extension/lib/garbage-collector.js

```diff
@@ -5,68 +5,25 @@
 "use strict";
 
 const { Cc, Ci, Cu } = require("chrome");
-const { EventEmitter } = require("sdk/deprecated/events");
+const { emit, on, off } = require("sdk/event/core");
 const prefs = require("sdk/preferences/service");
 const timers = require("sdk/timers");
 const unload = require("sdk/system/unload");
 
 const config = require("./config");
 
-Cu.import('resource://gre/modules/Services.jsm');
-
-const reporter = EventEmitter.compose({
-  _pref_gc_notifications: null,
-
-  constructor: function Reporter() {
-    // Report unhandled errors from listeners
-    this.on("error", console.exception.bind(console));
-
-    // Make sure we clean-up correctly
-    unload.ensure(this, 'unload');
-
-    if (config.application.branch >= 16) {
-      this._pref_gc_notifications = config.preferences.memory_notify;
-    }
-    else {
-      this._pref_gc_notifications = config.preferences.memory_log;
```

Comment:
Please also update the config.py and remove those entries which are not necessary anymore.